ui/src/pages/Results.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {Component} from 'react';
+import {FormattedMessage, defineMessages, injectIntl} from 'react-intl';
+import {withRouter} from 'react-router-dom';
+import Fab from '@material/react-fab';
+
+import Footer from './../components/Footer.js';
+import LoadingIndicator from './../components/LoadingIndicator.js';
+import RepoHeader from './../components/RepoHeader.js';
+import SearchBar from './../components/SearchBar.js';
+
+const messages = defineMessages({
+  createNewRecord: {
+    id: 'Results.createNewRecord',
+    defaultMessage: 'Create new record',
+    description: 'Label on a button for creating a new database entry.',
+  },
+  noResultsFound: {
+    id: 'Results.noResultsFound',
+    defaultMessage: 'No results found.',
+    description: 'A message shown when a search query has produced no results.',
+  },
+});
+
+/*
+ * A component for showing a search results page.
+ */
+class Results extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoaded: false,
+      error: null,
+      repo: null,
+      results: null
+    };
+    this.goToAdd = this.goToAdd.bind(this);
+    this.loadResults = this.loadResults.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+  }
+
+  goToAdd() {
+    this.props.history.push({
+          pathname: '/' + this.props.match.params.repoId + '/create',
+      });
+  }
+
+  loadResults() {
+    const query = new URL(window.location.href).searchParams.get('query_name');
+    var apiCalls = [
+        '/' + this.props.match.params.repoId + '/d/repoinfo',
+        '/' + this.props.match.params.repoId + '/d/results?query=' + query,
+        ];
+    Promise.all(apiCalls.map(url => fetch(url)))
+        .then(res => Promise.all(res.map(r => r.json())))
+        .then(
+          (res) => {
+            this.setState({
+              isLoaded: true,
+              repo: res[0],
+              results: res[1]
+            });
+          },
+          (error) => {
+            this.setState({error: error});
+          }
+        );
+  }
+
+  componentDidMount() {
+    this.loadResults();
+  }
+
+  handleSearch(query) {
+    this.setState({isLoaded: false});
+    this.props.history.push({
+        pathname: '/' + this.props.match.params.repoId + '/results',
+        search: '?query_name=' + query,
+      });
+    this.loadResults();
```

Comment:
I just imagined react-router handles this kind of thing automatically i.e., it automatically re-renders the page based on the new URL pushed to the history. Isn't it not the case?

Reply:
For the most part it does, but not in this case. From what I can tell, the reason is because react-router considers its responsibility to be strictly just the routing, and, since this is staying within the results page (the same route), it's not react-router's job.

Reply:
So it looks like componentWillReceiveProps was recently deprecated, and I think getDerivedStateFromProps is more complex than we need, but you're right that the prior code is pretty ugly. It turns out that if we get the query_name value from props.location.search (instead of from window) we can use componentDidUpdate, so I did that.